### PR TITLE
Feature/gbd 2017 paf update

### PR DESCRIPTION
**Pull request by michaella21**
raw validation part for the paf/etiology paf. It's not big, but confusing, try to add comments as much as possible, since most information about the relation among paf, exposure, rr are from James and can be very confusing to the others

### CODE REVIEW

File: src/vivarium_inputs/core.py [hunk 1]
```diff
@@ -167,8 +167,9 @@ def get_exposure(entity: Union[RiskFactor, AlternativeRiskFactor, CoverageGap],
 
     if entity.distribution in ['dichotomous', 'ordered_polytomous', 'unordered_polytomous']:
         tmrel_cat = sorted(list(entity.categories.to_dict()), key=lambda x: int(x[3:]))[-1]
-        exposed = data[~data.isin(tmrel_cat)]
-        unexposed = data[data.isin(tmrel_cat)]
+        exposed = data[~data.isin([tmrel_cat])]
+        unexposed = data[data.isin([tmrel_cat])]
```
Comment (halfhorst) [inline src/vivarium_inputs/core.py]: once you have one index, you can do data.index.difference(exposed.index). I think that will be much faster. isin() doesn't scale very well in my experience. I haven't directly compared, though

File: src/vivarium_inputs/core.py [hunk 2]
```diff
@@ -266,20 +267,18 @@ def get_relative_risk(entity: Union[RiskFactor, CoverageGap], location_id: int)
...
 def get_population_attributable_fraction(entity: Union[RiskFactor, Etiology], location_id: int) -> pd.DataFrame:
-    data = extract.extract_data(entity, 'population_attributable_fraction', location_id)
-
     if entity.kind == 'risk_factor':
-        restriction_entity = entity
+        data = extract.extract_data(entity, 'population_attributable_fraction', location_id)
+
     else:  # etiology
+        data = extract.extract_data(entity, 'etiology_population_attributable_fraction', location_id)
         cause = [c for c in causes if c.etiologies and entity in c.etiologies][0]
```
Comment (kfwilson) [inline src/vivarium_inputs/core.py]: just fyi I updated causes so now they have empty tuples for etiologies and sequelae so we shouldn't have to do the `if c.etiologies` anymore

File: src/vivarium_inputs/validation/raw.py [hunk 3]
```diff
@@ -7,8 +8,10 @@
 from gbd_mapping import (ModelableEntity, Cause, Sequela, RiskFactor,
                          Etiology, Covariate, CoverageGap, causes)
 
+
```
Comment (kfwilson) [inline src/vivarium_inputs/validation/raw.py]: extra return

File: src/vivarium_inputs/validation/raw.py [hunk 4]
```diff
@@ -936,7 +941,9 @@ def validate_relative_risk(data: pd.DataFrame, entity: Union[RiskFactor, Coverag
 
 
 def validate_population_attributable_fraction(data: pd.DataFrame, entity: Union[RiskFactor, Etiology],
-                                              location_id: int, estimation_years: pd.Series) -> None:
+                                              location_id: int, estimation_years: pd.Series,
+                                              relative_risk: pd.DataFrame, exposure: pd.DataFrame) -> None:
+
```
Comment (kfwilson) [inline src/vivarium_inputs/validation/raw.py]: It looks like this is a version before the doc strings came in. Could you merge that in and update the doc strings accordingly?

File: src/vivarium_inputs/validation/raw.py [hunk 5]
```diff
@@ -948,32 +955,69 @@ def validate_population_attributable_fraction(data: pd.DataFrame, entity: Union[
...
-    if entity.kind == 'risk_factor':
-        restrictions_entity = entity
-    else:  # etiology
-        restrictions_entity = [c for c in causes if c.etiologies and entity in c.etiologies][0]
+    restrictions = entity.restrictions
+    male_expected = not restrictions.female_only
+    female_expected = not restrictions.male_only
+
+    grouped = data.groupby(['cause_id', 'measure_id'], as_index=False)
+    grouped.apply(lambda df: check_age_group_ids(df, None, None))
```
Comment (kfwilson) [inline src/vivarium_inputs/validation/raw.py]: why do you need the lambda functions here? couldn't you just do `grouped.apply(check_age_group_ids, None, None)`?

File: src/vivarium_inputs/validation/raw.py [hunk 6]
```diff
@@ -948,32 +955,69 @@ def validate_population_attributable_fraction(data: pd.DataFrame, entity: Union[
...
+            raise DataAbnormalError(f'Paf data for {entity.kind} {entity.name} affecting {cause.name} contains yll '
+                                    f'values despite the affected entity being restricted to yld only.')
+        if cause.restrictions.yll_only and np.any(data[(data.cause_id == c_id) & (data.measure_id == MEASURES['YLDs'])]):
+            raise DataAbnormalError(f'Paf data for {entity.kind} {entity.name} affecting {cause.name} contains yld '
+                                    f'values despite the affected entity being restricted to yll only.')
+
+    grouped.apply(lambda df: check_paf_rr_exposure_age_groups(df, relative_risk, exposure, entity))
+
+
+def validate_etiology_population_attributable_fraction(data: pd.DataFrame, entity: Union[RiskFactor, Etiology],
```
Comment (kfwilson) [inline src/vivarium_inputs/validation/raw.py]: can you add a doc string here? also I think the typing info for entity should just be etiology

File: src/vivarium_inputs/validation/raw.py [hunk 7]
```diff
@@ -936,7 +941,9 @@ def validate_relative_risk(data: pd.DataFrame, entity: Union[RiskFactor, Coverag
 
 
 def validate_population_attributable_fraction(data: pd.DataFrame, entity: Union[RiskFactor, Etiology],
-                                              location_id: int, estimation_years: pd.Series) -> None:
```
Comment (kfwilson) [inline src/vivarium_inputs/validation/raw.py]: typing for entity should just be RiskFactor now right?

File: src/vivarium_inputs/validation/raw.py [hunk 8]
```diff
@@ -1383,6 +1426,85 @@ def check_cause_age_restrictions_sets(entity: Cause) -> None:
                                       f' We currently do not support these causes.')
 
 
+def check_paf_age_groups_cause_restrictions(paf: pd.DataFrame) -> None:
+    """Raise an error if relative risk includes any age_group_ids with non
+    trivial data. But paf still can have an extra data while there is no
+    relative risk matching with.
+    """
+
+    cause_map = {c.gbd_id: c for c in causes}
```
Comment (kfwilson) [inline src/vivarium_inputs/validation/raw.py]: why do you need to make this cause_map when you only want a single cause?

File: src/vivarium_inputs/validation/raw.py [hunk 9]
```diff
@@ -1383,6 +1426,85 @@ def check_cause_age_restrictions_sets(entity: Cause) -> None:
...
+            e_othercols = [c for c in exposure.columns if c not in DRAW_COLUMNS]
+            df = exposure.set_index(e_othercols)
+            op = operator.lt if entity.tmred.inverted else operator.gt
+            exposed_age_groups = set(df[op(df, tmrel)].reset_index().age_group_id)
+
+            valid_rr = valid_rr[valid_rr.age_group_id.isin(exposed_age_groups)]
+            rr_age_groups = set(valid_rr.age_group_id)
+
+        else:  # categorical distribution
+            #  Non-trivial rr for categorical risk factors is where relative risk is not equal to 1.
```
Comment (kfwilson) [inline src/vivarium_inputs/validation/raw.py]: thanks for all these comments! this is really helpful

File: src/vivarium_inputs/core.py (hunk 1 above)
Comment (halfhorst) [reply src/vivarium_inputs/core.py]: alternatively, comparison against a set is much faster, too

File: src/vivarium_inputs/validation/raw.py [hunk 10]
```diff
@@ -619,7 +621,8 @@ def validate_deaths(data: pd.DataFrame, entity: Cause,
         Location to which the data should pertain.
     estimation_years
         Expected set of years, used to check the `year_id` column in `data`.
-
+    population
+        Population data , used to check upper bound of `data`
```
Comment (kfwilson) [inline src/vivarium_inputs/validation/raw.py]: period please

File: src/vivarium_inputs/validation/raw.py [hunk 11]
```diff
@@ -935,8 +941,47 @@ def validate_relative_risk(data: pd.DataFrame, entity: Union[RiskFactor, Coverag
...
+    data
+        Population attributable fraction data for `entity` in location `location_id`.
+    entity
+        Risk factor to which the data pertain.
+    location_id
+        Location to which the data should pertain.
+    relative_risk
+        Relative risk data for 'entity' in location 'location_id'.
+    exposure
+        Exposure data for 'entity' in location 'location_id'
```
Comment (kfwilson) [inline src/vivarium_inputs/validation/raw.py]: period please

File: src/vivarium_inputs/validation/raw.py [hunk 12]
```diff
@@ -905,21 +913,19 @@ def validate_relative_risk(data: pd.DataFrame, entity: Union[RiskFactor, Coverag
...
-        grouped.apply(check_sex_restrictions, context, entity.restrictions.male_only, entity.restrictions.female_only)
         for (c_id, morb, mort, _), g in grouped:
             cause = [c for c in causes if c.gbd_id == c_id][0]
             if morb == 1:
                 start, end = cause.restrictions.yld_age_group_id_start, cause.restrictions.yld_age_group_id_end
             else:  # morb = 0 , mort = 1
                 start, end = cause.restrictions.yll_age_group_id_start, cause.restrictions.yll_age_group_id_end
             check_age_restrictions(g, context, start, end, error=False)
+            check_sex_ids(g, context, male_expected, female_expected, cause=cause)
+            check_sex_restrictions_paf_rr(g, context, restrictions, cause.restrictions)
```
Comment (collijk) [inline src/vivarium_inputs/validation/raw.py]: If risk/cause pair in protective:
hard lower at 0, soft upper at 1, hard upper at MAX_RR
else:
hard lower at 1, hard upper at max_rr

Comment (collijk) [reply src/vivarium_inputs/validation/raw.py]: @line 936, swap max val calculation

Comment (collijk) [reply src/vivarium_inputs/validation/raw.py]: Update docstring about sex restriction check

File: src/vivarium_inputs/validation/raw.py [hunk 13]
```diff
@@ -933,6 +939,45 @@ def validate_relative_risk(data: pd.DataFrame, entity: Union[RiskFactor, Coverag
 
 def validate_population_attributable_fraction(data: pd.DataFrame, entity: Union[RiskFactor, Etiology],
                                               context: RawValidationContext) -> None:
+
+    """Check the standard set of validations on raw population attributable
```
Comment (collijk) [inline src/vivarium_inputs/validation/raw.py]: Update for sex restrictions

File: src/vivarium_inputs/validation/raw.py [hunk 14]
```diff
@@ -933,6 +939,45 @@ def validate_relative_risk(data: pd.DataFrame, entity: Union[RiskFactor, Coverag
...
+        any values in columns do not match the expected set of values, or
+        yll/yld data exist for yld only/yll only causes.(or data exist outside
+        of cause age restrictions or data do not exist for the age groups for
+        which both exposure and relative risk exist.
+    """
+
+    protective_causes = PROTECTIVE_CAUSE_RISK_PAIRS[entity.name] if entity.name in PROTECTIVE_CAUSE_RISK_PAIRS else []
+
+    protective = data[data.cause_id.isin(protective_causes)]
+    non_protective = data.loc[data.index.difference(protective.index)]
```
Comment (collijk) [inline src/vivarium_inputs/validation/raw.py]: Move these to where they are used.

File: src/vivarium_inputs/validation/raw.py [hunk 15]
```diff
@@ -944,10 +989,80 @@ def validate_population_attributable_fraction(data: pd.DataFrame, entity: Union[
...
+
+    grouped = data.groupby(['cause_id', 'measure_id'])
+    grouped.apply(check_age_group_ids, context, None, None)
+    for (c_id, _), g in grouped:
+        cause = [c for c in causes if c.gbd_id == c_id][0]
+        check_sex_ids(g, context, male_expected, female_expected, cause=cause)
+        check_sex_restrictions_paf_rr(g, context, restrictions, cause.restrictions)
+
+    if not protective.empty:
+        check_value_columns_boundary(data, MIN_PROTECTIVE_PAF, 'lower', value_columns=DRAW_COLUMNS, inclusive=True,
```
Comment (collijk) [inline src/vivarium_inputs/validation/raw.py]: Use protective in args

File: src/vivarium_inputs/validation/raw.py [hunk 16]
```diff
@@ -944,10 +989,80 @@ def validate_population_attributable_fraction(data: pd.DataFrame, entity: Union[
...
-    else:  # etiology
-        restrictions_entity = [c for c in causes if c.etiologies and entity in c.etiologies][0]
+    restrictions = entity.restrictions
+    male_expected = not restrictions.female_only
+    female_expected = not restrictions.male_only
+
+    grouped = data.groupby(['cause_id', 'measure_id'])
+    grouped.apply(check_age_group_ids, context, None, None)
+    for (c_id, _), g in grouped:
+        cause = [c for c in causes if c.gbd_id == c_id][0]
```
Comment (collijk) [inline src/vivarium_inputs/validation/raw.py]: Update male expected and female expected here and pass in args

File: src/vivarium_inputs/validation/raw.py [hunk 17]
```diff
@@ -944,10 +989,80 @@ def validate_population_attributable_fraction(data: pd.DataFrame, entity: Union[
...
+        check_sex_ids(g, context, male_expected, female_expected, cause=cause)
+        check_sex_restrictions_paf_rr(g, context, restrictions, cause.restrictions)
```
Comment (collijk) [inline src/vivarium_inputs/validation/raw.py]: Refactor args for usage with normal function (must have male only or female only)

File: src/vivarium_inputs/validation/raw.py [hunk 18]
```diff
@@ -944,10 +989,80 @@ def validate_population_attributable_fraction(data: pd.DataFrame, entity: Union[
...
+
+    if not protective.empty:
+        check_value_columns_boundary(data, MIN_PROTECTIVE_PAF, 'lower', value_columns=DRAW_COLUMNS, inclusive=True,
+                                     error=DataAbnormalError)
+        check_value_columns_boundary(data, MIN_PAF, 'upper', value_columns=DRAW_COLUMNS, inclusive=True)
+        check_value_columns_boundary(data, MAX_PAF, 'upper', value_columns=DRAW_COLUMNS, inclusive=True,
+                                     error=DataAbnormalError)
+
+    check_value_columns_boundary(non_protective, MIN_PAF, 'lower', value_columns=DRAW_COLUMNS, inclusive=True,
```
Comment (collijk) [inline src/vivarium_inputs/validation/raw.py]: if not non_protective.empty: 
    Do these things.

File: src/vivarium_inputs/validation/raw.py [hunk 19]
```diff
@@ -944,10 +989,80 @@ def validate_population_attributable_fraction(data: pd.DataFrame, entity: Union[
...
+    for c_id in set(data.cause_id):
+        cause = [c for c in causes if c.gbd_id == c_id][0]
+        if cause.restrictions.yld_only and np.any(data[(data.cause_id == c_id) & (data.measure_id == MEASURES['YLLs'])]):
+            raise DataAbnormalError(f'Paf data for {entity.kind} {entity.name} affecting {cause.name} contains yll '
+                                    f'values despite the affected entity being restricted to yld only.')
+        if cause.restrictions.yll_only and np.any(data[(data.cause_id == c_id) & (data.measure_id == MEASURES['YLDs'])]):
+            raise DataAbnormalError(f'Paf data for {entity.kind} {entity.name} affecting {cause.name} contains yld '
+                                    f'values despite the affected entity being restricted to yll only.')
+
+    grouped.apply(check_paf_rr_exposure_age_groups, context, entity)
```
Comment (collijk) [inline src/vivarium_inputs/validation/raw.py]: Move up with sex restrictions check

File: src/vivarium_inputs/validation/raw.py [hunk 20]
```diff
@@ -944,10 +989,80 @@ def validate_population_attributable_fraction(data: pd.DataFrame, entity: Union[
...
+    expected_columns = ['metric_id', 'measure_id', 'rei_id', 'cause_id'] + DRAW_COLUMNS + DEMOGRAPHIC_COLUMNS
+    check_columns(expected_columns, data.columns)
+
+    check_measure_id(data, ['YLLs', 'YLDs'], single_only=False)
+    check_metric_id(data, 'percent')
+
+    check_years(data, context, 'annual')
+    check_location(data, context)
+
+    restrictions_entity = [c for c in causes if c.etiologies and entity in c.etiologies][0]
```
Comment (collijk) [inline src/vivarium_inputs/validation/raw.py]: Remove if c.etiologies

File: src/vivarium_inputs/validation/raw.py [hunk 21]
```diff
@@ -944,10 +989,80 @@ def validate_population_attributable_fraction(data: pd.DataFrame, entity: Union[
...
+                                     error=DataAbnormalError)
+
+    check_value_columns_boundary(non_protective, MIN_PAF, 'lower', value_columns=DRAW_COLUMNS, inclusive=True,
+                                 error=DataAbnormalError)
+    check_value_columns_boundary(non_protective, MAX_PAF, 'upper', value_columns=DRAW_COLUMNS, inclusive=True,
+                                 error=DataAbnormalError)
+
+    for c_id in set(data.cause_id):
+        cause = [c for c in causes if c.gbd_id == c_id][0]
+        if cause.restrictions.yld_only and np.any(data[(data.cause_id == c_id) & (data.measure_id == MEASURES['YLLs'])]):
```
Comment (collijk) [inline src/vivarium_inputs/validation/raw.py]: Extract function for use by both paf validators

File: src/vivarium_inputs/validation/raw.py [hunk 22]
```diff
@@ -1557,7 +1771,8 @@ def check_age_group_ids(data: pd.DataFrame, context: RawValidationContext,
 
 
 def check_sex_ids(data: pd.DataFrame, context: RawValidationContext,
-                  male_expected: bool = True, female_expected: bool = True, combined_expected: bool = False) -> None:
+                  male_expected: bool = True, female_expected: bool = True, combined_expected: bool = False,
+                  cause: Cause = None) -> None:
```
Comment (collijk) [inline src/vivarium_inputs/validation/raw.py]: Remove cause

File: src/vivarium_inputs/validation/raw.py [hunk 23]
```diff
@@ -1731,6 +1952,32 @@ def check_sex_restrictions(data: pd.DataFrame, context: RawValidationContext, ma
                                         'values for both males and females.')
 
 
+def check_sex_restrictions_paf_rr(data: pd.DataFrame, context: RawValidationContext, restrictions: Restrictions,
```
Comment (collijk) [inline src/vivarium_inputs/validation/raw.py]: Call from above

File: src/vivarium_inputs/validation/raw.py [hunk 24]
```diff
@@ -1368,6 +1484,104 @@ def check_mort_morb_flags(data: pd.DataFrame, yld_only: bool, yll_only: bool) ->
...
+    age_restrictions = {'YLLs': (cause.restrictions.yll_age_group_id_start, cause.restrictions.yll_age_group_id_end),
+                        'YLDs': (cause.restrictions.yld_age_group_id_start, cause.restrictions.yld_age_group_id_end)}
+    valid_rr = rr[(rr.cause_id == cause_id) & rr_measures[measure]]
+
+    # It means we have YLL Paf but mortality = morbidity = 1 and we do not support this case.
+    if measure == 'YLLs' and valid_rr.empty:
+        pass
+
+    else:
+        if entity.distribution in ['ensemble', 'lognormal', 'normal']:
```
Comment (collijk) [inline src/vivarium_inputs/validation/raw.py]: get_rr_age_groups function (think it only needs entity and context)

File: src/vivarium_inputs/validation/raw.py [hunk 25]
```diff
@@ -1368,6 +1484,104 @@ def check_mort_morb_flags(data: pd.DataFrame, yld_only: bool, yll_only: bool) ->
...
+            df = rr.set_index(rr_othercols)
+            rr_age_groups = set(df[df != 1].reset_index().age_group_id)
+
+        #  We apply the narrowest restrictions among exposed_age_groups, rr_age_gruops and cause age restrictions.
+        #  We may have paf outside of exposure/rr but inside of cause age restrictions, then warn it.
+        #  If paf does not exist for the narrowest range of exposure/rr/cause, raise an error.
+        cause_age_start, cause_age_end = age_restrictions[measure]
+        cause_restriction_ages = set(get_restriction_age_ids(cause_age_start, cause_age_end, age_group_ids))
+
+        age_start = max(min(rr_age_groups), cause_age_start)
```
Comment (collijk) [inline src/vivarium_inputs/validation/raw.py]: get intersection of rr and cause ages for use on line 1570

File: src/vivarium_inputs/core.py [hunk 26]
```diff
@@ -269,22 +271,44 @@ def get_relative_risk(entity: Union[RiskFactor, CoverageGap], location_id: int)
     return data
 
 
-def get_population_attributable_fraction(entity: Union[RiskFactor, Etiology], location_id: int) -> pd.DataFrame:
-    data = extract.extract_data(entity, 'population_attributable_fraction', location_id)
+def filter_by_relative_risk(df: pd.DataFrame, relative_risk: pd.DataFrame) -> pd.DataFrame:
+    c_id = df.cause_id.unique()[0]
+    rr = relative_risk[relative_risk.cause_id == c_id]
+    if set(rr.mortality) == {1} and set(rr.morbidity) == {1}:
```
Comment (collijk) [inline src/vivarium_inputs/core.py]: Inline comment why we're doing this. We presume all attributable mortality moves through incidence.

File: src/vivarium_inputs/validation/sim.py [hunk 27]
```diff
@@ -340,18 +341,35 @@ def validate_relative_risk(data: pd.DataFrame, entity: Union[RiskFactor, Coverag
...
-    check_value_columns_boundary(data, boundary_value=VALID_PAF_RANGE[0],
+    if not protective.empty:
+        check_value_columns_boundary(protective, boundary_value=VALID_PROTECTIVE_PAF_MIN, boundary_type='lower',
+                                     value_columns=['value'], error=DataTransformationError)
+        check_value_columns_boundary(protective, boundary_value=VALID_PAF_RANGE[0], boundary_type='upper',
+                                     value_columns=['value'])
+        check_value_columns_boundary(protective, boundary_value=VALID_PAF_RANGE[1], boundary_type='upper',
+                                     value_columns=['value'], error=DataTransformationError)
+
+    check_value_columns_boundary(non_protective, boundary_value=VALID_PAF_RANGE[0],
```
Comment (collijk) [inline src/vivarium_inputs/validation/sim.py]: if not non_protective.empty:
    do this stuff

File: src/vivarium_inputs/validation/sim.py [hunk 28]
```diff
@@ -340,18 +341,35 @@ def validate_relative_risk(data: pd.DataFrame, entity: Union[RiskFactor, Coverag
...
                                               context: SimulationValidationContext):
+    protective_causes = PROTECTIVE_CAUSE_RISK_PAIRS[entity.name] if entity.name in PROTECTIVE_CAUSE_RISK_PAIRS else []
+    protective = data[data.affected_entity.isin(protective_causes)]
+    non_protective = data.loc[data.index.difference(protective.index)]
+
     risk_relationship = data.groupby(['affected_entity', 'affected_measure'])
     risk_relationship.apply(validate_standard_columns, context)
 
-    check_value_columns_boundary(data, boundary_value=VALID_PAF_RANGE[0],
+    if not protective.empty:
```
Comment (collijk) [inline src/vivarium_inputs/validation/sim.py]: Move protective split after standard col validators

**Comment by collijk** [comment]
rr split between protective and non-protective in validators

File: src/vivarium_inputs/validation/sim.py [hunk 29]
```diff
@@ -315,16 +316,22 @@ def validate_relative_risk(data: pd.DataFrame, entity: Union[RiskFactor, Coverag
...
-                                 error=DataTransformationError)
-    check_value_columns_boundary(data, boundary_value=VALID_RELATIVE_RISK_RANGE[1][range_kwd],
-                                 boundary_type='upper', value_columns=['value'],
-                                 error=DataTransformationError)
+    protective_causes = PROTECTIVE_CAUSE_RISK_PAIRS[entity.name] if entity.name in PROTECTIVE_CAUSE_RISK_PAIRS else []
+    protective = data[data.affected_entity.isin([c.name for c in protective_causes])]
+    non_protective = data.loc[data.index.difference(protective.index)]
+
+    if not protective.empty:
+        check_value_columns_boundary(data, boundary_value=0, boundary_type='lower',
```
Comment (kfwilson) [inline src/vivarium_inputs/validation/sim.py]: pass protective not data and in next call too

File: src/vivarium_inputs/validation/sim.py [hunk 30]
```diff
@@ -315,16 +316,22 @@ def validate_relative_risk(data: pd.DataFrame, entity: Union[RiskFactor, Coverag
...
+                                     value_columns=['value'], inclusive=False,
+                                     error=DataTransformationError)
+        check_value_columns_boundary(data, boundary_value=VALID_RELATIVE_RISK_RANGE[0],
+                                     boundary_type='upper', value_columns=['value'])
+    if not non_protective.empty:
+        check_value_columns_boundary(data, boundary_value=VALID_RELATIVE_RISK_RANGE[0],
```
Comment (kfwilson) [inline src/vivarium_inputs/validation/sim.py]: pass non_protective not data

**Pull request merged**